pym/bob/input.py

```diff
@@ -1737,6 +1737,21 @@ def validate(self, data):
                 None)
         return data
 
+class LayerValidator:
+    def __init__(self):
+        self.__scmValidator = ScmValidator({
+            'git' : GitScm.SCHEMA,
+            'svn' : SvnScm.SCHEMA,
+            'cvs' : CvsScm.SCHEMA,
+            'url' : UrlScm.SCHEMA})
+
+    def validate(self, data):
+        if 'name' not in data:
+            raise schema.SchemaMissingKeyError("Missing 'name' key in {}".format(data), None)
+        name = data.get('name')
+        del data['name']
```

Comment:
You should copy `data` before. Otherwise you're changing the parsed result which could lead to side effects.